### PR TITLE
[Validator] refactoring IsTrueValidator

### PR DESCRIPTION
**Pull request by sfmok**
| Q             | A
| ------------- | ---
| Branch?       | 6.2 
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| License       | MIT
| Refactoring? | yes

### CODE REVIEW

File: src/Symfony/Component/Validator/Constraints/IsTrueValidator.php [hunk 1]
```diff
@@ -29,15 +29,13 @@ public function validate(mixed $value, Constraint $constraint)
             throw new UnexpectedTypeException($constraint, IsTrue::class);
         }
 
-        if (null === $value) {
+        if (null === $value || true === $value || 1 === $value || '1' === $value) {
```
Comment (zenas1210) [inline src/Symfony/Component/Validator/Constraints/IsTrueValidator.php:32]: maybe in_array would look better?

Comment (ayyoub-afwallah) [reply src/Symfony/Component/Validator/Constraints/IsTrueValidator.php:32]: with  $strict = true

**Comment by derrabus** [comment]
The change is technically correct, but I don't think that it adds any value, tbh.

**Comment by sfmok** [comment]
> The change is technically correct, but I don't think that it adds any value, tbh.

Yes, I just added this update to follow the same style code as in IsFalseValidator

**Review by derrabus — APPROVED** [review]
> Yes, I just added this update to follow the same style code as in IsFalseValidator

That sounds reasonable to me. 🙂

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @sfmok.

**Pull request merged**